integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -1604,37 +1611,39 @@ fn test_gc_fork_tail() {
 
 #[test]
 fn test_tx_forwarding() {
-    let mut genesis_config = GenesisConfig::test(Clock::real());
-    genesis_config.epoch_length = 100;
-    let mut env = TestEnv::builder(&genesis_config)
-        .clients_count(50)
-        .validator_seats(50)
-        .mock_epoch_managers()
-        .build();
-    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
-    let genesis_hash = *genesis_block.hash();
-    // forward to 2 chunk producers
+    let mut genesis = Genesis::test(TestEnvBuilder::make_accounts(50), 50);
+    genesis.config.epoch_length = 100;
+    let mut env =
+        TestEnv::builder_from_genesis(&genesis).clients_count(50).validator_seats(50).build();
+
+    let signer = InMemorySigner::test_signer(&"test0".parse().unwrap());
+    let tx = env.tx_from_actions(vec![], &signer, signer.get_account_id());
+
+    // client_index is picked so that with epoch_info rng tx wouldn't need to be forwarded to
+    // client itself.
+    let client_index = 1;
     assert_eq!(
-        env.clients[0].process_tx(SignedTransaction::empty(genesis_hash), false, false),
+        env.clients[client_index].process_tx(tx, false, false),
         ProcessTxResponse::RequestRouted
     );
-    assert_eq!(env.network_adapters[0].requests.read().unwrap().len(), 4);
+    assert_eq!(
+        env.network_adapters[client_index].requests.read().unwrap().len(),
+        env.clients[client_index].config.tx_routing_height_horizon as usize
+    );
 }
 
 #[test]
 fn test_tx_forwarding_no_double_forwarding() {
-    let mut genesis_config = GenesisConfig::test(Clock::real());
-    genesis_config.epoch_length = 100;
-    let mut env = TestEnv::builder(&genesis_config)
-        .clients_count(50)
-        .validator_seats(50)
-        .mock_epoch_managers()
-        .build();
-    let genesis_block = env.clients[0].chain.get_block_by_height(0).unwrap();
-    let genesis_hash = *genesis_block.hash();
+    let mut genesis = Genesis::test(TestEnvBuilder::make_accounts(50), 50);
+    genesis.config.epoch_length = 100;
+    let mut env =
+        TestEnv::builder_from_genesis(&genesis).clients_count(50).validator_seats(50).build();
+
+    let signer = InMemorySigner::test_signer(&"test0".parse().unwrap());
+    let tx = env.tx_from_actions(vec![], &signer, signer.get_account_id());
     // The transaction has already been forwarded, so it won't be forwarded again.
     assert_eq!(
-        env.clients[0].process_tx(SignedTransaction::empty(genesis_hash), true, false),
+        env.clients[0].process_tx(tx, /*is_forwarded*/ true, false),
```

Comment:
nit: I suggest introducing a variable instead: `let is_forwarded = true` and then using it

Reply:
Done in [568a366](https://github.com/near/nearcore/pull/13097/commits/568a366b2840cc9496d103c3aac1ee31879ad740)